bokeh/io.py

```diff
@@ -390,19 +391,29 @@ def save(obj, filename=None, resources=None, title=None, state=None, validate=Tr
 
 def _detect_filename(ext):
     """ Detect filename from the name of the script being run. Returns
-    None if the script could not be found (e.g. interactive mode).
+    temporary file if the script could not be found or the location of the
+    script does not have write permission (e.g. interactive mode).
     """
     import inspect
-    from os.path import isfile, dirname, basename, splitext, join
+    from os.path import dirname, basename, splitext, join, curdir
 
     frame = inspect.currentframe()
     while frame.f_back and frame.f_globals.get('name') != '__main__':
         frame = frame.f_back
 
     filename = frame.f_globals.get('__file__')
-    if filename and isfile(filename):
-        name, _ = splitext(basename(filename))
-        return join(dirname(filename), name + "." + ext)
+
+    if filename is None:
+        return tempfile.NamedTemporaryFile().name
+
+    # dirname(filename) returns empty str if file called from within directory
```

Comment:
I am just trying to check the wherever the file will get written to that dir is writeable. So if dirname(filename) returns empty string then we need to check that the current directory is writeable.